src/Airship/Internal/Route.hs

```diff
@@ -80,14 +80,19 @@ star = Route [RestUnbound]
 newtype RoutingSpec m a = RoutingSpec { getRouter :: Writer [(Route, Resource m)] a }
     deriving (Functor, Applicative, Monad, MonadWriter [(Route, Resource m)])
 
-route :: [(Route, a)] -> [Text] -> a -> (a, (HashMap Text Text, [Text]))
-route routes pInfo resource404 = foldr' (matchRoute pInfo) (resource404, (mempty, mempty)) routes
+route :: [(Route, a)]
+      -> [Text]
+      -> Maybe (a, (HashMap Text Text, [Text]))
```

Comment:
There also might be more expressive alternatives to folds we can think of to build routing primitives. `mapAccumL` is a good one. This could all be addressed in another PR though.